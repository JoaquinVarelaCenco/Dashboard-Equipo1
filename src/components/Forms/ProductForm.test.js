import { render, screen, act } from "@testing-library/react";

describe("Test formulario de productos", ()=>{

    test("Formulario", ()=>{})
// crear mockup producto

    //validar inputs
    //validar botones
    //validar agregar | quitar imágenes
    //validar función de cantidad (agregar | quitar stock)

    //agregar | cancelar -> en caso de addProduct
    //editar  | cancelar | inputs con value -> en caso de editProduct
  
})