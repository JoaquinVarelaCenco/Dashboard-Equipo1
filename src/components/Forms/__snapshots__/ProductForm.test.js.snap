// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test formulario de productos Se renderiza el componente 1`] = `
<div>
  <div
    class="product-new"
  >
    <div
      class="product-new__view-card"
    >
      <img
        alt="imagen producto"
        src="no-image.jpeg"
      />
      <div
        class="product-new__view-card__content"
      >
        <h2 />
        <div
          class="product-new__view-card__info"
        >
          <div>
             
            <p>
              0
            </p>
            <span>
              PUNTOS SUPERCLUB
            </span>
          </div>
          <div>
             
            <p>
              0
            </p>
            <span>
              STOCK DISPONIBLE
            </span>
             
          </div>
          <div
            class="profile-container"
          >
            <a
              class="eachLink-view undefined"
              href="/stores"
            >
              <button
                class="user-button-view"
              >
                <div
                  class="user-button__profile-view"
                >
                  <img
                    class="user-img-view"
                    src="ProfilePic.png"
                  />
                </div>
                <div>
                  <p
                    class="user-name-view"
                  >
                    Olivia store
                  </p>
                </div>
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
    <h1>
      Informacion
    </h1>
    <form
      class="product-new__form"
    >
      <p
        class="product-new__input-title"
      >
        Nombre
      </p>
      <input
        minlength="3"
        name="title"
        placeholder="Nombre"
        required=""
        type="text"
        value=""
      />
      <p
        class="product-new__input-title"
      >
        Valor
      </p>
      <input
        min="0"
        name="price"
        required=""
        type="number"
        value="0"
      />
      <p
        class="product-new__input-title"
      >
        Stock
      </p>
      <div
        class="product-new-form__stock"
      >
        <button
          class="noHoverEffect"
          type="button"
        >
          -
        </button>
        <span
          data-testid="spanProdStock"
        >
          0
        </span>
        <button
          class="noHoverEffect"
          type="button"
        >
          +
        </button>
      </div>
      <p
        class="product-new__input-title"
      >
        Descripcion
      </p>
      <textarea
        name="description"
        placeholder="Descripcion del producto"
        required=""
        type="number"
      />
      <p
        class="product-new__input-title"
      >
        Categoria
      </p>
      <select
        class="form-control"
        name="category"
        required=""
      >
        <option
          disabled=""
          value=""
        >
          Seleccione categoria...
        </option>
        <option
          value="categoria-1"
        >
          categoria-1
        </option>
        <option
          value="categoria-2"
        >
          categoria-2
        </option>
        <option
          value="categoria-3"
        >
          categoria-3
        </option>
        <option
          value="categoria-4"
        >
          categoria-4
        </option>
      </select>
      <p
        class="product-new__input-title"
      >
        Tienda
      </p>
      <select
        class="form-control"
        name="store"
      >
        <option
          disabled=""
          selected=""
          value=""
        >
          Seleccione una tienda
        </option>
        <option
          value="tienda-1"
        >
          tienda-1
        </option>
        <option
          value="tienda-2"
        >
          tienda-2
        </option>
        <option
          value="tienda-3"
        >
          tienda-3
        </option>
        <option
          value="tienda-4"
        >
          tienda-4
        </option>
      </select>
      <h3>
        Galeria de imagenes
      </h3>
      <p
        class="product-new__input-title"
      >
        Nueva imagen
      </p>
      <div
        class="prueba"
      >
        <input
          id="input-img-add"
          name="image"
          type="text"
        />
        <button
          type="button"
        >
          Cargar
        </button>
      </div>
      <p
        class="product-new__input-title"
      >
        Imagenes actuales
      </p>
      <div
        class="product-new__action"
      >
        <button
          type="submit"
        >
          Editar
        </button>
        <button
          type="reset"
        >
          Cancelar
        </button>
      </div>
    </form>
  </div>
</div>
`;
